feat: provide PageProps, LayoutProps types (#13308)
sveltejs/kit: documentation/docs/20-core-concepts/10-routing.md

```svelte
<!--- file: src/routes/blog/[slug]/+page.svelte --->
<script>

> 	/** @type {{ data: import('./$types').PageProps }} */

Reviewer: ```suggestion
	/** @type {import('./$types').PageProps} */
```

```

> [!LEGACY]

> > `PageProps` was added in 2.16.0. In earlier versions, you can type the `data` property manually with `PageData` instead, see [$types](#\$types).

Reviewer: ```suggestion
> `PageProps` was added in 2.16.0. In earlier versions, you had to type the `data` property manually with `PageData` instead, see [$types](#\$types).
```

{@render children()}
```

> [!LEGACY]

> > `LayoutProps` was added in 2.16.0. In earlier versions, you can [type the properties manually instead](#\$types).

Reviewer: ```suggestion
> `LayoutProps` was added in 2.16.0. In earlier versions, you had to [type the properties manually instead](#\$types).
```

Throughout the examples above, we've been importing types from a `$types.d.ts` file. This is a file SvelteKit creates for you in a hidden directory if you're using TypeScript (or JavaScript with JSDoc type annotations) to give you type safety when working with your root files.

> For example, annotating `let { data } = $props()` with `PageProps` tells TypeScript that the type of `data` is whatever was returned from `load`:

Reviewer: ```suggestion
For example, annotating `let { data } = $props()` with `PageProps` (or `LayoutProps`, for a `+layout.svelte` file) tells TypeScript that the type of `data` is whatever was returned from `load`:
```

```svelte
<!--- file: src/routes/blog/[slug]/+page.svelte --->
<script>
	/** @type {import('./$types').PageProps} */
	let { data } = $props();
</script>
```

> [!NOTE]

> > The `PageProps` and `LayoutProps` types, added in 2.16.0, are a shortcut for typing the `data` prop as `PageData` or `LayoutData`, as well as other props, such as `form` for pages, or `children` for layouts. In earlier versions, you can type these properties manually, for example for a page:

Reviewer: ```suggestion
> The `PageProps` and `LayoutProps` types, added in 2.16.0, are a shortcut for typing the `data` prop as `PageData` or `LayoutData`, as well as other props, such as `form`, or `children` for layouts. In earlier versions, you had to type these properties manually, for example for a page:
```

>

> > ```js
> > /// file: +page.svelte
> > /** @type {{ data: import('./$types').PageData, form: import('./$types').ActionData }} */
> > let { data, form } = $props();
> > ```

Reviewer: Should these be Svelte code blocks instead so that they're the same as the other examples?
```suggestion
> ```svelte
> <!--- file: +page.svelte --->
> <script>
> 	  /** @type {import('./$types').PageProps} */
>   let { data, form } = $props();
> </script>
> ```
```

Author: I didn't to make the code blocks a little smaller, this of course can be changes. The idea of it is to just repeat the part of the larger code block above that needs to be changed.

Reviewer 2: Let's keep it small and not change it
